cmd/clusterctl/client/repository/repository_versions.go

```diff
@@ -109,28 +106,50 @@ func latestPatchRelease(ctx context.Context, repo Repository, major, minor *uint
 			continue
 		}
 
-		// track prereleases separately
-		if sv.PreRelease() != "" {
-			if latestPrereleaseVersion == nil || latestPrereleaseVersion.LessThan(sv) {
-				latestPrereleaseTag = v
-				latestPrereleaseVersion = sv
-			}
-			continue
-		}
+		parsedReleaseVersions = append(parsedReleaseVersions, sv)
+	}
 
-		if latestReleaseVersion == nil || latestReleaseVersion.LessThan(sv) {
-			latestTag = v
-			latestReleaseVersion = sv
-		}
+	versionCandidates := parsedReleaseVersions
+
+	if len(parsedReleaseVersions) == 0 {
+		return "", errors.New("failed to find releases tagged with a valid semantic version number")
 	}
 
-	// Fall back to returning latest prereleases if no release has been cut or bail if it's also empty
-	if latestTag == "" {
-		if latestPrereleaseTag == "" {
-			return "", errors.New("failed to find releases tagged with a valid semantic version number")
+	// Sort parsed versions by semantic version order.
+	sort.SliceStable(versionCandidates, func(i, j int) bool {
+		// Prioritize release versions over pre-releases. For example v1.0.0 > v2.0.0-alpha
```

Comment:
Nah... It's obvious that v2.0.0 > v2.0.0-alpha. Here I wanted to emphasise we always prioritize versions. So, even if v1.0.0 is less than v2.0.0-alpha by semver, we pick v1 anyway because it is a release.